### PR TITLE
Best Buy Ruby post

### PR DESCRIPTION
**Pull request by SandroDamilano**
#### :link: Issue reference:

* [Best Buy Ruby, the wrapper gem for accessing the Best Buy APIs](https://github.com/rootstrap/blog/issues/101)

---

#### Description:

Have you tried out Best Buy APIs? Would you like to have an enhanced experience when using it? If so, let me introduce to you one of our newest gems: Best Buy Ruby.

---

#### :pushpin: Notes:

- The title may be too long. I'm open to ideas
- Am I leaving anything out?
- Is it necessary to add any summary?

---

#### 🗒️ ToDo:

- [x] Double check post title
- [x] Add cover image

---

#### :file_folder: Categories

Backend, Ruby

---
#### :mag: Keywords:

- Best Buy
- ruby
- gem
- API

### CODE REVIEW

File: content/Access the world's most powerful retail APIs with Best Buy Ruby - Sandro Damilano.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+BestBuy::Products.new.get_by(
+    category_id: some_id,
+    item_contidion: 'new',
+    pagination: { page: 5, page_size: 100 }
+)
+```
+
+### _Great, I want it. Where can I find it?_
+
+The gem is available in [RubyGems](https://rubygems.org/gems/best_buy_ruby), but be sure to check our own [Github repo](https://github.com/rootstrap/best_buy_ruby), where you'll find the docs to learn how to use it. But of course anyone is welcomed to submit any issues or contribute!
```
Comment (juanmanuelramallo) [inline content/Access the world's most powerful retail APIs with Best Buy Ruby - Sandro Damilano.md:43]:
```suggestion
The gem is available in [RubyGems](https://rubygems.org/gems/best_buy_ruby), but be sure to check our own [Github repo](https://github.com/rootstrap/best_buy_ruby), where you'll find the docs to learn how to use it. You are welcomed to submit any issues or contribute!
```

File: content/Access the world's most powerful retail APIs with Best Buy Ruby - Sandro Damilano.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+## Access the world's most powerful retail APIs with Best Buy Ruby
+
+![Main image](images/blue_led_connection.jpg)
```
Comment (juanmanuelramallo) [inline content/Access the world's most powerful retail APIs with Best Buy Ruby - Sandro Damilano.md:3]: Shouldn't we ask Sofi for help on the cover image?

File: content/Access the world's most powerful retail APIs with Best Buy Ruby - Sandro Damilano.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+### _What's Best Buy Ruby?_
+
+Best Buy Ruby is a new gem developed entirely here, at Rootstrap. It's basically a wrapper for the different [Best Buy APIs](https://bestbuyapis.github.io/api-documentation/). It offers a simple and intuitive interface that allows users to access the required API with as much customizations as they like.
+
+### _Why use it?_
+
+Best Buy offers a really powerful API. The different options provided range from pagination and sorting methods to incredibly flexible item queries and response formats.
+
+But _with great power, there must also come great responsibility_.
```
Comment (santib) [inline content/Access the world's most powerful retail APIs with Best Buy Ruby - Sandro Damilano.md:13]:
```suggestion
But _with great power comes great responsibility_.
```

File: content/Access the world's most powerful retail APIs with Best Buy Ruby - Sandro Damilano.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+Most healthy projects working with this API must build some kind of module to manage these options, even when the amount of interactions made with the API are low.
+
+And that's where best_buy_ruby comes to save the day.
+
+It's built entirely around the goal of making the API both easily accessible and fully customizable at the same time.
+
+Let's see a quick example:
+
+With the gem, this request:
```
Comment (santib) [inline content/Access the world's most powerful retail APIs with Best Buy Ruby - Sandro Damilano.md:25]:
```suggestion
Let's see a quick example. With the gem, this request:
```

File: content/Access the world's most powerful retail APIs with Best Buy Ruby - Sandro Damilano.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+https://api.bestbuy.com/v1/products(categoryPath.id=some_id&(condition=new|new=true))?format=json&page=5&pageSize=100&apiKey=SOME_API_KEY
+```
+
+Can be written like this:
+
+```ruby
+BestBuy::Products.new.get_by(
+    category_id: some_id,
+    item_contidion: 'new',
+    pagination: { page: 5, page_size: 100 }
```
Comment (santib) [inline content/Access the world's most powerful retail APIs with Best Buy Ruby - Sandro Damilano.md:37]:
```suggestion
  category_id: some_id,
  item_contidion: 'new',
  pagination: { page: 5, page_size: 100 }
```

File: content/Access the world's most powerful retail APIs with Best Buy Ruby - Sandro Damilano.md (hunk 3 above)
Comment (SandroDamilano) [reply content/Access the world's most powerful retail APIs with Best Buy Ruby - Sandro Damilano.md:13]: I just thought that fancy version was funnier hahaha https://en.wikipedia.org/wiki/With_great_power_comes_great_responsibility
But I sure don't mind changing it!

File: content/Access the world's most powerful retail APIs with Best Buy Ruby - Sandro Damilano.md (hunk 2 above)
Comment (SandroDamilano) [reply content/Access the world's most powerful retail APIs with Best Buy Ruby - Sandro Damilano.md:3]: I have no idea haha do we?

File: content/Access the world's most powerful retail APIs with Best Buy Ruby - Sandro Damilano.md (hunk 3 above)
Comment (santib) [reply content/Access the world's most powerful retail APIs with Best Buy Ruby - Sandro Damilano.md:13]: I didn't know that version was also valid, so I'm ok 👌

Comment (SandroDamilano) [reply content/Access the world's most powerful retail APIs with Best Buy Ruby - Sandro Damilano.md]: Well, that's actually a very valid point. If that version might not be very well known, let's just go with the common one!

**Pull request merged**